common/lib/xmodule/xmodule/partitions/tests/test_partitions.py

```diff
@@ -325,9 +325,19 @@ def test_get_user_group_id_for_partition(self):
 
         # switch to the second group and verify that it is returned for the user
         self.user_partition.scheme.current_group = groups[1]    # pylint: disable=no-member
-        group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #group2_id = self.partition_service.get_user_group_id_for_partition(user_partition_id)
+        #self.assertEqual(group2_id, groups[1].id)    # pylint: disable=no-member
+
+        partition_service2 = StaticPartitionService(
+            [self.user_partition],
+            user=Mock(username='ma', email='ma@edx.org', is_staff=False, is_active=True),
+            course_id=self.course.id,
+            track_function=Mock()
+        )
+        group2_id = partition_service2.get_user_group_id_for_partition(user_partition_id)
```

Comment:
Sorry, I hadn't seen your message when I wrote that.

So I can definitely make the default behavior non-caching unless an explicit cache object is passed in. This thing is only created once in our app, in the `LmsModuleSystem` constructor. I still have the same question, though perhaps pushed one level up the stack. It looks like the pieces that are doing the real work (`get_group_for_user` in `RandomUserPartitionScheme` and `CohortPartitionScheme`) will both set a value if there exists None beforehand. I'm assuming that the values they spit back within a given request are going to stay the same. I *think* that's safe, but I'm really not familiar with this part of the code.